docs/architecture/adr-001-node-key-refactoring.md

```diff
@@ -95,27 +95,15 @@ We can migrate nodes one by one by iterating the version.
 
 We assume keeping only the range versions of `fromVersion` to `toVersion`. Refer to [this issue](https://github.com/cosmos/cosmos-sdk/issues/12989).
 
-When we want to prune all versions up to the specific version `n`
+Here we are introducing a new way how to get orphaned nodes which remove in the `n+1`th version updates.
 
-- Iterate the tree based on the root of `n+1`th version.
-- Iterate the node until visiting the node the version is below `fromVersion` and don't visit further deeply.
-- Apply `DeletePath` for all visited nodes the version is below `n+1`.
+- Traverse the tree in-order way based on the root of `n+1`th version.
+- If we visit the lower version node, pick the node and don't visit further deeply. Pay attention to the order of these nodes.
+- Traverse the tree in-order way based on the root of `n`th version.
```

Comment:
https://github.com/cosmos/iavl/issues/635
I proposed a method to prune version by traversing two version simutanously, it can be applied to current iavl tree, don't need the path as nonce.

Reply:
it looks similar with the above suggestion, I agree with you, the `getting orphans` actually is not required a `nonce` or `path`.
I think we can create a separate PR. @tac0turtle what about you?

Reply:
makes sense.